### PR TITLE
fix reset mask rnn target generator #716

### LINKED ISSUE #716: Fine-tuning and train on my own dataset
Hello, 
I'm doing instance segmentation for a specific data. I prepared my own dataset similar to coco format because gluoncv just support coco and my dataset contain just 3 classes, I trained my data from scratch and it works. I would like to do a fine-tuning because I don't have enough data and not to encounter overfitting. Is there any way to replace the output layer of the pre-trained model "mask_rcnn_resnet50_v1b_coco" to fit the right number of classes for my own images? @zhreshold

### PR DESCRIPTION
**Pull request by zhreshold**
Fix #716

### CODE REVIEW

File: gluoncv/model_zoo/mask_rcnn/mask_rcnn.py [hunk 1]
```diff
@@ -173,8 +173,9 @@ def __init__(self, features, top_features, classes, mask_channels=256, rcnn_max_
                 roi_size = (self._roi_size[0] * 2, self._roi_size[1] * 2)
             else:
                 roi_size = self._roi_size
+            self._roi_size = roi_size
             self.mask_target = MaskTargetGenerator(
-                self._max_batch, self._num_sample, self.num_class, roi_size)
+                self._max_batch, self._num_sample, self.num_class, self._roi_size)
```
Comment (Jerryzcn) [inline gluoncv/model_zoo/mask_rcnn/mask_rcnn.py]: the roi size here is actually different than roi size used in other places thus it is not assigned to self._roi_size. maybe a field is needed. When deep_fcn is used with FPN, it upsamples the output size to 2x the original typically 28x28, where as the default does not use FPN, which keeps the original 14x14 size.

**Comment by mli** [comment]
Job PR-733-3 is done. 
Docs are uploaded to http://gluon-vision-staging.s3-website-us-west-2.amazonaws.com/PR-733/3/index.html 
Code coverage of this PR:  vs. Master:
